analytics/management/commands/create_ga_report.py

```diff
@@ -0,0 +1,325 @@
+                value = data['value']
+                try:
+                    text = value['text']
+                except KeyError:
+                    word_count_stats[type] += 1
+                    continue
+                except TypeError:
+                    text = value
+                # 'List' type...
+                if isinstance(text, list):
+                    text = '\n'.join(text)
+                words = self._get_words(text)
+                word_count_stats[type] += len(words)
+        # 'Chapters'...
+        try:
+            chapters = page.chapters
+        except AttributeError:
+            chapters = []
+        if len(chapters) > 0:
+            type = 'ChapterBody'
+            text_keys = ['label', 'text']
+            word_count_stats[type] = 0
+            for chapter in page.chapters:
+                # Recurse down the 'body' tree and collect all the 'text'...
+                text = '\n'.join(self._get_body_text(chapter.value['body'], text_keys))
+                words = self._get_words(text)
+                word_count_stats[type] += len(words)
+        return word_count_stats
+
+    def _get_body_text(self, body, text_keys):
+        texts = []
+        for body_block in body:
+            if isinstance(body_block.value, dict):
+                if 'body' in body_block.value:
+                    text = '\n'.join(self._get_body_text(body_block.value['body'], text_keys))
+                else:
+                    for key in text_keys:
+                        if key in body_block.value:
+                            try:
+                                text = body_block.value[key].source
+                            except AttributeError:
+                                text = body_block.value[key]
+                    for key in body_block.value.keys():
+                        if key not in ['body', 'expandable', 'heading_level', 'image', 'include_border', 'label', 'placement', 'text', 'use_dropcap']:
+                            raise KeyError("'{}' not in the list of expected keys for a StreamField".format(key))
+            elif isinstance(body_block.value, list):
+                text = '\n'.join([x.source for x in body_block.value])
+            else:
+                try:
+                    text = body_block.value.source
+                except AttributeError:
+                    text = body_block.value
+            texts.append(text)
+        return texts
+
+    def _get_words(self, text):
+        # Strip out MS-Word specific tags and other junk...probably don't have all of it, but this should handle most...
+        pattern = '<o:DocumentProperties>.*</o:DocumentProperties>|\{mso.*;\}'
+        text, replacements = re.subn(pattern, '', text, flags=re.DOTALL)
+        # Strip HTML tags
+        pattern = '<.*?>'
+        text, replacements = re.subn(pattern, '', text, flags=re.DOTALL)
+        words = re.findall(ur'\w+', text.replace(u'\xa0', ' ').encode('utf-8'))
```

Comment:
It a big topic, you should google it, but the up shot is unicode is a universal encoding and utf-8 is a specific way one could encode text. Generally you should decode incoming bytes to unicode strings assuming you know the encoding, and then encode those unicode strings to your prefer output encoding when you write data out.